packages/web-pkg/src/composables/actions/spaces/useSpaceActionsDuplicate.ts

```diff
@@ -0,0 +1,152 @@
+import { SpaceResource } from '@ownclouders/web-client'
+import { computed } from 'vue'
+import { SpaceAction, SpaceActionOptions } from '../types'
+import { useGettext } from 'vue3-gettext'
+import { useStore } from '../../store'
+import { useAbility } from '../../ability'
+import { useClientService } from '../../clientService'
+import { useLoadingService } from '../../loadingService'
+import { Store } from 'vuex'
+import { buildSpace, isProjectSpaceResource } from '@ownclouders/web-client/src/helpers'
+import { Drive } from '@ownclouders/web-client/src/generated'
+import { resolveFileNameDuplicate } from '../../../helpers'
+import PQueue from 'p-queue'
+
+export const useSpaceActionsDuplicate = ({
+  store
+}: {
+  store?: Store<any>
+} = {}) => {
+  store = store || useStore()
+  const { $gettext } = useGettext()
+  const ability = useAbility()
+  const clientService = useClientService()
+  const loadingService = useLoadingService()
+
+  const duplicateSpace = async (existingSpace: SpaceResource) => {
+    const projectSpaces: SpaceResource[] = store.getters['runtime/spaces/spaces'].filter(
+      (space: SpaceResource) => isProjectSpaceResource(space)
+    )
```

Comment:
Since the method is now being called multiple times I think we should make this a computed property for performance reasons.

Reply:
Ah, good point. It's probably even desired to fetch the spaces each time to avoid name conflicts with newly created ones.